releases/README.md

```diff
@@ -42,21 +42,27 @@
 3. Push tag and Github release
 
    1. Push the tag to `git@github.com:containerd/containerd.git`.
-      NOTE: this will kick off CI building of the release binaries.
+      NOTE: this will kick off CI to prepare the Github release and binaries.
 
-   2. Create the Github release using the `Tag version` which was just pushed. Use the first
-      line outputted from the release tool as the `Release title` and the remainder of the
-      output for the description. No alteration of the release output should be needed.
-      Ensure `pre-release` is checked if an `-rc`.
-      NOTE: This should be done immediately after pushing the tag, otherwise CI may create the release
-      when the binaries are pushed.
+   2. Verify release job has started in the actions tab and wait for release job to complete
+      successfully. If the job fails, attempt to restart it. If the job continuously fails,
+      delete the tag from GitHub as soon as possible and fix the issue before restarting the
+      release process.
 
-   3. Check CI has completed and uploaded the binaries. Remove any binaries which get
-      uploaded but are not intended as part of the release (e.g. Darwin binaries).
+   3. Check the Github release, ensure the content matches what is expected and all the
+      release binaries are included.
```

Comment:
Could you consider signing binaries? (for >= 2.0)
https://github.com/containerd/nerdctl/pull/2089/files#diff-ed933e266827bac3e1788519c912361cfaa862541ab38663e867b0a3978c34ca

Reply:
Definitely could consider it. The focus of this PR is how it has been done for existing releases. We can add to the process then update this doc. What is pointed to there is a bit complicated and probably something we should try and automate from the dev end (maybe in release tool?)

Reply:
We can try it for the 2.0 beta releases first, and if it works out we can update this doc. I don't want to add steps we haven't actually done in a release yet.